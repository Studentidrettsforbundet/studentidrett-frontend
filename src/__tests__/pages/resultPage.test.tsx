import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ResultPage from '../../pages/resultPage';
import configureMockStore from 'redux-mock-store';

const store = configureMockStore();

const mockProps = {
    state: {
        recommendation: [
            {
                name: 'name1',
                id: 'test1',
            },
            {
                name: 'name2',
                id: 'test2',
            },
            {
                name: 'name3',
                id: 'test3',
            },
        ],
    },
};

describe('resultPage', () => {
    describe('snapshot tests', () => {
        it('should render correctly', () => {
            const tree = renderer
                .create(
                    <Provider store={store({})}>
                        <BrowserRouter>
                            <ResultPage location={mockProps} />
                        </BrowserRouter>
                        ,
                    </Provider>,
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
