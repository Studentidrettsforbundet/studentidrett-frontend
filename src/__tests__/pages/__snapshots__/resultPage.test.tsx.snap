// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resultPage snapshot tests should render correctly 1`] = `
Array [
  <div
    className="fkhz08q"
  >
    <h1>
      Idrettsvalgomat
    </h1>
    <p>
      Dine topp 3 resultater er:
    </p>
    <a
      className="unstyled_link"
      href="/sports/test1"
      onClick={[Function]}
    >
      <div
        className="fcjpwci"
      >
        <div
          className="f1q24w9w"
        >
          <h5
            className="f19aa3jt"
          >
            name1
          </h5>
          <img
            alt="Sport"
            className="f1mqlstf"
            src="placeholder.png"
          />
        </div>
      </div>
    </a>
    <a
      className="unstyled_link"
      href="/sports/test2"
      onClick={[Function]}
    >
      <div
        className="fcjpwci"
      >
        <div
          className="f1q24w9w"
        >
          <h5
            className="f19aa3jt"
          >
            name2
          </h5>
          <img
            alt="Sport"
            className="f1mqlstf"
            src="placeholder.png"
          />
        </div>
      </div>
    </a>
    <a
      className="unstyled_link"
      href="/sports/test3"
      onClick={[Function]}
    >
      <div
        className="fcjpwci"
      >
        <div
          className="f1q24w9w"
        >
          <h5
            className="f19aa3jt"
          >
            name3
          </h5>
          <img
            alt="Sport"
            className="f1mqlstf"
            src="placeholder.png"
          />
        </div>
      </div>
    </a>
    <br />
    <a
      className="f1yhg9g btn btn-primary"
      href="/questionnaire"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Ta valgomaten igjen
    </a>
  </div>,
  ",",
]
`;
